src/components/PopupWithYes.js

```diff
@@ -0,0 +1,26 @@
+import Popup from './Popup.js';
+
+export default class PopupWithYes extends Popup{
+    constructor({popupSelector, handleYesButton}){
+        super({popupSelector});
+        this._handleYesButton = handleYesButton;
+        this._popupButton = this._popupElement.querySelector('.modal__button')
+    }
+
+    setEventListeners(){
+        super.setEventListeners();
+
+        this._popupElement.querySelector('.modal__close-button-container').addEventListener('click', () => {
+            this._popupButton.removeEventListener('click', deleteCard);
+        });
+
+        const deleteCard = (evt) => {
+            evt.preventDefault();
+            this._handleYesButton();
+            this.close();
+            this._popupButton.removeEventListener('click', deleteCard);
+        }
+
+        this._popupButton.addEventListener('click', deleteCard);
```

Comment:
This code is incorrect

Please, consult your mentor about the logics

You add a  listener in line 13 every time you call `setEventListeners` which causes a memory leak


`.setEventListeners()` should be removed from `openings` because it's a **public** method that should be called only **once** in `index.js` otherwise you add listeners again and again to the same elements. This can cause a **`memory leak`**

Reply:
You haven't corrected my remark https://github.com/Reynaldito98/se_project_aroundtheus/pull/9#discussion_r1385346564